homeassistant/components/switch/pwrctrl.py

```diff
@@ -0,0 +1,125 @@
+"""
+Support for ANEL PwrCtrl switches.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.pwrctrl/
+"""
+import logging
+import socket
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.switch import (SwitchDevice, PLATFORM_SCHEMA)
+from homeassistant.const import (CONF_HOST, CONF_PASSWORD, CONF_USERNAME)
+from homeassistant.util import Throttle
+
+
+REQUIREMENTS = ['https://github.com/mweinelt/anel-pwrctrl/archive/'
+                'master.zip#anel_pwrctrl==0.0.1']
+
+CONF_PORT_RECV = "port_recv"
+CONF_PORT_SEND = "port_send"
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=5)
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_PORT_RECV): cv.port,
+    vol.Required(CONF_PORT_SEND): cv.port,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_HOST): cv.string,
+})
+
+
+# pylint: disable=unused-variable
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup PwrCtrl devices/switches."""
+    host = config.get(CONF_HOST, None)
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    port_recv = int(config.get(CONF_PORT_RECV))
```

Comment:
If you want it to be an int, add to your `PLATFORM_SCHEMA` the following instead of `cv.string`: `vol.Coerce(int)`